### PR TITLE
Clear signals before setting

### PR DESCRIPTION
**Pull request by alisonbelow**
This solves the bug where a signal was not correctly able to be set more than once (depending on values) in the derived Frame class.

When setting a signal, the buffer was always `|=` and therefore any bit that was set high remained high when setting the next signal. Now signal will be cleared prior to setting each time.

See new unit test for fixed functionality.

### CODE REVIEW

File: cantools/database/can/templates/DBC.h [hunk 1]
```diff
@@ -98,6 +99,15 @@ class Frame {
     return oss.str();
   }
 
+  std::string BinaryString() const {
```
Comment (alisonbelow) [inline cantools/database/can/templates/DBC.h:102]: Used for debugging, potentially useful for developer

**Comment by alisonbelow** [comment]
I also am aware this is a fairly severe bug and I apologize for not fixing it sooner!

**Review by SaildroneDave — APPROVED** [review]
Looks good, thanks for fixing!
The clear_* methods are implicitly tested with the new test_SetSignalsMultipleTimes_Vehicle_DBC, but might want to have an explicit test for one of them?
Good to have a test for the new BinaryString, partly to demonstrate functionality

**Review by Dillonmcewan — APPROVED** [review]
LGTM!

**Review by eneojustin — APPROVED** [review]

**Pull request merged**